### ISSUE TITLE
Implement Theme Manager

### ISSUE THREAD

**Issue by YazeedAlKhalaf** [Stacked-Org/stacked#128]
Hello Dane @FilledStacks ,

I am writing this issue after I sent you an email. What I would like to propose is implementing a theme service that can be used to theme the app easily.

Guidelines: (Can be changed, but before starting with the work)

- [ ] It must be a service
- [ ] It must have the ability to expand to multiple themes, not just dark and light. 
- [ ] An intuitive way where the developer can run:
     - [ ] A method for adding a theme (custom by app user)
     - [ ] A method for setting the theme
     - [ ] A method for getting the current theme
- [ ] Implementation should be in a way that it is easy so that the theme argument accepts a value from the service and the user can change the theme easily without getting out of the app.

It would be a great enhancement 🚀

**Comment by FilledStacks** [comment]
Hi @YazeedAlKhalaf 

Service classes is for performing functionality or exposing functionality in a `ViewModel`. Themes is UI only functionality to it should exist in the UI only and not be available through the locator. 

I am planning on making a Theme manager using stacked that's [exactly this implementation](https://www.filledstacks.com/snippet/build-a-theme-manager-in-flutter/) wrapped up with different names.

**Comment by YazeedAlKhalaf** [comment]
That's Great!
Can we use this issue to follow the progress?

**Comment by FilledStacks** [comment]
I would probably post updates here or forget and just publish the new package 😄

**Comment by YazeedAlKhalaf** [comment]
@FilledStacks  I just wanna note that the implementation should be:

* You can add multiple themes easily.
* Dark and Light should be the two default, may be mandatory 😅.
* There should be a follow system color function.

Thanks 🚀

**Comment by FilledStacks** [comment]
High demands haha. I'll probably get to this much later. Maybe I'll ask one of the other devs in the company to implement it. I'm currently swamped with client work and none of them need themes at the moment so this is low priority.

**Comment by FilledStacks** [comment]
Progress of the theme functionality can be seen [here](https://github.com/FilledStacks/stacked/tree/master/packages/stacked_themes).

- [x] Multiple theme functionality
- [x] Last selected theme persistence
- [x] Status bar color update functionality
- [ ] Dark and Light Theme functionality selected by System
- [ ] Dark and Light Theme functionality selected by User
- [ ] Persist selected Dark/Light theme when selected by User
- [ ] Create service to access ThemeManager functionality from ViewModels

The example in the lib has the first three tasks implemented to see how it can be used.

**Comment by FilledStacks** [comment]
The [first version](https://pub.dev/packages/stacked_themes) of stacked themes has been published. Busy with the recording for the new tutorial. If you have any issues or feature requests please don't hesitate to file an issue or a PR (PR would be appreciated over an issue :) )
